src/Neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -199,10 +199,24 @@ private static TransactionAttribute[] DeserializeAttributes(ref MemoryReader rea
             int count = (int)reader.ReadVarInt((ulong)maxCount);
             TransactionAttribute[] attributes = new TransactionAttribute[count];
             HashSet<TransactionAttributeType> hashset = new();
+            Dictionary<TransactionAttributeType, int> typeCount = new();
             for (int i = 0; i < count; i++)
             {
                 TransactionAttribute attribute = TransactionAttribute.DeserializeFrom(ref reader);
-                if (!attribute.AllowMultiple && !hashset.Add(attribute.Type))
+                if (typeCount.TryGetValue(attribute.Type, out int tc))
+                {
+                    tc++;
+
+                    if (attribute.AllowedCount < tc)
```

Comment:
Use `MaxTransactionAttributes`?

Reply:
Is already checked in the for condition (`count`)